### PR TITLE
Remove native Windows support

### PR DESCRIPTION
**Pull request by dcbw**
Remove the ability to run ovn-kubernetes natively on Windows. This
hasn't worked for a while as ovnkube has required OVN 2.11 for the
past couple months, which isn't available on Windows that we know
of.

Note that this does not affect the Windows hybrid-overlay code and
binary. That is still supported.

@girishmg @danwinship

### CODE REVIEW

**Comment by dcbw** [comment]
@girishmg note that this doesn't yet optimize the MAC address reading stuff that required the annotation dance.

For that, did we decide that the management port and localnet gateway can use a pre-determined MAC address (based off IP) and avoid the node annotation dance?

**Comment by girishmg** [comment]
> @girishmg note that this doesn't yet optimize the MAC address reading stuff that required the annotation dance.
> 
> For that, did we decide that the management port and localnet gateway can use a pre-determined MAC address (based off IP) and avoid the node annotation dance?

Yes, that was what we agreed to.

**Comment by girishmg** [comment]
@dcbw overall looks good (Except for the build error). I think the following check can go as well

```
func runningPlatform() (string, error) {
	if runtime.GOOS == windowsOS {
		return windowsOS, nil
	}
```

File: go-controller/pkg/config/config.go [hunk 1]
```diff
@@ -180,7 +179,7 @@ type CNIConfig struct {
 	ConfDir string `gcfg:"conf-dir"`
 	// Plugin specifies the name of the CNI plugin
 	Plugin string `gcfg:"plugin"`
-	// Windows ONLY, specifies the ID of the HNS Network to which the containers will be attached
+	// UNUSED; remove in the near future
 	WinHNSNetworkID string `gcfg:"win-hnsnetwork-id"`
```
Comment (danwinship) [inline go-controller/pkg/config/config.go:183]: Is there any reason to keep this? No one on Linux should be trying to set it

File: go-controller/pkg/config/config.go [hunk 2]
```diff
@@ -1151,14 +1150,8 @@ func getConfigFilePath(ctx *cli.Context) (string, bool) {
...
 	}
-
 	// Linux default
-	if runtime.GOOS != "windows" {
-		return filepath.Join("/etc", "openvswitch", "ovn_k8s.conf"), true
-	}
-
-	// Windows default
-	return filepath.Join(os.Getenv("OVS_SYSCONFDIR"), "ovn_k8s.conf"), true
+	return filepath.Join("/etc", "openvswitch", "ovn_k8s.conf"), true
```
Comment (danwinship) [inline go-controller/pkg/config/config.go:1154]: `return "/etc/openvswitch/ovn_k8s.conf"`

File: go-controller/pkg/node/gateway_init.go [hunk 3]
```diff
@@ -172,9 +171,7 @@ func CleanupClusterNode(name string) error {
 	}
 
 	// Delete iptable rules for management port on Linux.
```
Comment (danwinship) [inline go-controller/pkg/node/gateway_init.go:173]: drop "on Linux"

File: go-controller/pkg/config/config.go (hunk 1 above)
Comment (dcbw) [reply go-controller/pkg/config/config.go:183]: Yeah.

File: go-controller/pkg/config/config.go (hunk 2 above)
Comment (dcbw) [reply go-controller/pkg/config/config.go:1160]: done

File: go-controller/pkg/node/gateway_init.go (hunk 3 above)
Comment (dcbw) [reply go-controller/pkg/node/gateway_init.go]: done

**Comment by dcbw** [comment]
@danwinship comments accepted/addressed, PTAL thanks!

**Pull request merged**